components/tidb_query/src/rpn_expr/impl_cast.rs

```diff
@@ -1052,7 +1052,9 @@ mod tests {
         assert!(r.is_none());
     }
 
-    fn make_ctx_about_overflow_truncate_should_clip_to_zero(
+    /// Rust has no overload, so we has `make_ctx_1`, `make_ctx_2`, etc..
+    /// Please call one of these functions according to the param list.
+    fn make_ctx_1(
```

Comment:
How about adding structure
```rust
struct Ctx1Options {
        overflow_as_warning: bool,
        truncate_as_warning: bool,
        should_clip_to_zero: bool,
}
```

It is more clear.

Reply:
Yes, in rust, this way is best.
```rust
pub struct ConfigFlag {
    overflow_as_warning: bool,
    truncate_as_warning: bool,
    should_clip_to_zero: bool,
    in_insert_stmt: bool,
    in_select_stmt: bool,
}

impl Default for ConfigFlag {
    fn default() -> Self {
        ConfigFlag {
            overflow_as_warning: false,
            truncate_as_warning: false,
            should_clip_to_zero: false,
            in_insert_stmt: false,
            in_select_stmt: false,
        }
    }
}

impl From<ConfigFlag> for EvalConfig {
    fn from(c: ConfigFlag) -> Self {
        let flag = Flag::empty();
        flag |= Flag::OVERFLOW_AS_WARNING * c.overflow_as_warning as u64;
        flag |= Flag::OVERFLOW_AS_WARNING * c.truncate_as_warning as u64;
        flag |= Flag::OVERFLOW_AS_WARNING * c.should_clip_to_zero as u64;
        flag |= Flag::OVERFLOW_AS_WARNING * c.in_insert_stmt as u64;
        flag |= Flag::OVERFLOW_AS_WARNING * c.in_select_stmt as u64;
        EvalConfig { flag, max_warning_cnt: 1024 }
    }
}

#[test]
fn test_1() {
    let ctx = EvalContext {
        cfg: Arc::new(EvalConfig::from(ConfigFlag {
            overflow_as_warning: true,
            ..ConfigFlag::default()
        })),
        ..EvalContext::default()
    };
    
    // test code
}
```
However, if refactoring all unit tests using this way, the work is not small.

BTW, if in other language that has no `..Xx::default()`, we has only two ways

- One
   ```rust
   #[test]
   fn test_2() {
       let mut flag: Flag = Flag::empty();
       flag |= Flag::OVERFLOW_AS_WARNING;
       let cfg = Arc::new(EvalConfig::from_flag(flag));
       let ctx = EvalContext::new(cfg);
   
       // test code
   }
   ```
- Two 
   ```rust
   fn make_ctx_1(overflow_as_warning: bool) -> EvalContext {
       let mut flag: Flag = Flag::empty();
       flag |= Flag::OVERFLOW_AS_WARNING;
       let cfg = Arc::new(EvalConfig::from_flag(flag));
       EvalContext::new(cfg)
   }
   
   #[test]
   fn test_3() {
       let ctx = make_ctx_1(true);
       // test code
   }
   ```

The second way is better, because the first one will repeat the same code in many unit test functions, it is very easy to make mistake if repeating code. 

However, how to name the `make_ctx_x`? If in Java, we can call all of them `make_ctx`. And in rust, we can simulate `overload` using `make_ctx_1`, `make_ctx_2`, etc., the caller need to call it according to `funcName + paramList`

Reply:
For languages that support named parameters, I think using named parameters is the best. The solution I posted here is a simulation for the named parameter.